docs/src/modules/components/OpenMuiChat.tsx

```diff
@@ -0,0 +1,209 @@
+  '--color-3': '210 100% 63%',
+  '--color-4': '195 100% 63%',
+  '--color-5': '90 100% 63%',
+  '--text-color': '#fff',
+
+  position: 'relative',
+  display: 'inline-flex',
+  height: 26,
+  padding: '7px 8px 8px 8px', // 7px for optical alignment
+  flexShrink: 0,
+  borderRadius: 999,
+  border: '1px solid transparent',
+  borderColor: alpha(theme.palette.grey[200], 0.8),
+  borderBottomColor: 'transparent',
+  fontSize: theme.typography.pxToRem(13),
+  fontWeight: theme.typography.fontWeightMedium,
+  color: theme.palette.primary[600],
+  '& .MuiSvgIcon-root': {
+    color: theme.palette.primary.main,
+  },
+  backgroundSize: '200%',
+  backgroundClip: 'padding-box, border-box, border-box',
+  backgroundOrigin: 'border-box',
+  animation: `${rainbow} 3s linear infinite`,
+  transition: 'background-color 0.2s, color 0.2s',
+  cursor: 'pointer',
+  '--bg-color-raw': '255, 255, 255',
+  '--bg-color': 'rgb(var(--bg-color-raw))',
+  backgroundImage: `linear-gradient(var(--bg-color), var(--bg-color)), linear-gradient(var(--bg-color) 50%, rgba(var(--bg-color-raw), 0.6) 80%, rgba(var(--bg-color-raw), 0)), linear-gradient(90deg, hsl(var(--color-1)), hsl(var(--color-5)), hsl(var(--color-3)), hsl(var(--color-4)), hsl(var(--color-2)))`,
+
+  ...theme.applyDarkStyles({
+    '--bg-color-raw': '0, 0, 0',
+    color: theme.palette.primary[300],
+    borderColor: alpha(theme.palette.primary[300], 0.2),
+    '& .MuiSvgIcon-root': {
+      color: theme.palette.primary[300],
+    },
+  }),
+
+  '&::before': {
+    content: '""',
+    position: 'absolute',
+    bottom: '-20%',
+    left: '50%',
+    zIndex: 0,
+    height: '20%',
+    width: '60%',
+    transform: 'translateX(-50%)',
+    animation: `${rainbow} 3s linear infinite`,
+    background:
+      'linear-gradient(90deg, hsl(var(--color-1)), hsl(var(--color-5)), hsl(var(--color-3)), hsl(var(--color-4)), hsl(var(--color-2)))',
+    filter: 'blur(0.8rem)',
+  },
+
+  '& > svg': {
+    height: 12,
+    width: 12,
+    margin: '1px 4px 0 4px',
+  },
+  '& > svg > path': {
+    fill: 'currentColor',
+  },
+  '& .MuiCircularProgress-root': {
+    color: 'var(--text-color)',
+  },
+}));
+
+const productToPackage: Record<string, string> = {
+  'material-ui': '@mui/material',
+  'joy-ui': '@mui/joy',
+  'x-data-grid': '@mui/x-data-grid',
+  'x-date-pickers': '@mui/x-date-pickers',
+  'x-tree-view': '@mui/x-tree-view',
+  'x-charts': '@mui/x-charts',
+};
+
+export default function OpenInMUIChatButton(props: OpenInMUIChatButtonProps) {
+  const { ...otherProps } = props;
+  const { productIdentifier } = React.use(PageContext);
+  const router = useRouter();
+  const productId = getProductInfoFromUrl(router.asPath).productId;
+  const packageName = productToPackage[productId];
+  const packageVersion =
+    (
+      productIdentifier.versions.find((it) => it.current)?.text ??
+      productIdentifier.versions?.[0]?.text
+    ) // slice to remove the 'v' prefix
+      ?.slice(1) ?? 'latest';
```

Comment:
This logic seems broken. Shouldn't it use the same one as CodeSandbox and StackBlitz?

Reply:
Updated to reuse those logic.